identity cache: Support looking up reserved identities

tgraf opened the pull request:
* Support resolving and listing reserved identities
 * Simplify bpf/init.sh by hardcoding the reserved identities
   which are static.
 * Convert `cilium identity list` and `cilium identity get` to
   use `text/tabwriter` and add uniformal `-o json` support.
 * Always list reserved identities for `cilium identity list`

Example:
```
$ cilium identity list
ID      LABELS
4       [reserved:health]
3       [reserved:cluster]
1       [reserved:host]
2       [reserved:world]
58770   [reserved:health]
12536   [container:id.bar]
2558    [container:id.foo]
```

Fixes: #2857

joestringer reviewed (approved):
LGTM, but would be nice to see a green build. Current failure appears unrelated.

tgraf committed:
cache: Support looking up reserved identities

 * Support resolving and listing reserved identities
 * Simplify bpf/init.sh by hardcoding the reserved identities
   which are static.
 * Convert `cilium identity list` and `cilium identity get` to
   use `text/tabwriter` and add uniformal `-o json` support.
 * Always list reserved identities for `cilium identity list`

Example:
```
$ cilium identity list
ID      LABELS
4       [reserved:health]
3       [reserved:cluster]
1       [reserved:host]
2       [reserved:world]
58770   [reserved:health]
12536   [container:id.bar]
2558    [container:id.foo]
```

Fixes: #2857

Signed-off-by: Thomas Graf <thomas@cilium.io>

manalibhutiyani commented on bpf/init.sh:
```diff
@@ -323,7 +321,7 @@ elif [ "$MODE" = "lb" ]; then
 		echo 1 > /proc/sys/net/ipv6/conf/all/forwarding
 
-		CALLS_MAP="cilium_calls_lb_${ID}"
+		CALLS_MAP="cilium_calls_lb"
```
Should this be `CALLS_MAP="cilium_calls_lb_${ID_HOST}"` or ID_WORLD?

tgraf replied:
No, LB is always a single program, this was a copy&paste error but it didn't have any negative effect.

aanm commented on pkg/identity/cache.go:
```diff
@@ -0,0 +1,128 @@
+	}
+
+	if gi, ok := allocatorKey.(globalIdentity); ok {
+		return NewIdentity(id, gi.Labels)
```
Why don't we store the updated identity from the KVStore into the cache?

tgraf replied:
The lookup is using the cache in the allocator but falls back to kvstore if it can't be found in the cache. I'm just moving the code here, I'm not adding or modifying code.

joestringer reviewed (approved):
LGTM for bpf / cli.
